microinvoice.test.js

```diff
@@ -0,0 +1,17 @@
+const Microinvoice = require('./lib/index');
+
+
+jest.mock('./lib/index', () => ({
```

Comment:
We don't want to mock the main function itself, this really doesn't test anything. 

Ideally we want to make a call create an instance of the MicroInvoice class as it would be used by someone. I would recommend first before you even write a test to use the MicroInvoice package as if you were going to create an invoice. So create a invoice manually to get a feel for how the package is susposed to function, and then create a test. 

Mocks are typically used when there is some external resource that should be mocked out which if it wasn't would make the test "nondeterministic" or require a system to be stood up. 

Some examples of things you typically mock for unit tests in web applications: 

- Databases
- External HTTP servers
- Authentication Systems

Since we make no external HTTP calls or need to standup any external systems for this package we don't really need to mock anything.

Reply:
Here is an example of using the micro invoice  package. You'll need to figure out some of the replaceable `items` / `payload`, EG this won't work if you simply copy and paste it, but it should get you started. 

```ts
        const parts = items.map(({ name, quantity, unit_cost }) => [
            { value: name },
            { value: quantity },
            {
                value: unit_cost.toString(),
                price: true,
                digits: Invoice.digitsEnforcer(unit_cost),
            },
            { value: (quantity * unit_cost).toFixed(2), price: true },
        ]);
        const legal = [
            terms
                ? {
                      value: `Terms ${terms}`,
                      weight: 'bold',
                      color: 'primary',
                  }
                : undefined,
            {
                value: `Powered by Paigo`,
                color: 'secondary',
            },
        ].filter((e) => e);
        const invoice = new MicroInvoice({
            style: {
                header: {
                    backgroundColor: '#FFFFFF',
                    height: 200,
                    image: logo
                        ? {
                              path: logo,
                              width: 100,
                          }
                        : undefined,
                },
            },
            data: {
                invoice: {
                    name: 'Invoice',
                    header: [
                        {
                            label: 'Invoice Number',
                            value: payload.number,
                        },
                        {
                            label: 'Invoice Date',
                            value: payload.date,
                        },
                        {
                            label: 'Due Date',
                            value: payload.due_date,
                        },
                        {
                            label: 'Balance Due',
                            value: payload.balanceDue.toFixed(2),
                            price: true,
                        },
                    ],
                    currency: payload.currency,
                    customer: [
                        {
                            label: 'Bill To',
                            value: [to],
                        },
                    ],
                    seller: [
                        {
                            label: 'Bill From',
                            value: [from],
                        },
                    ],
                    details: {
                        header: [
                            {
                                value: 'Item',
                            },
                            {
                                value: 'Quantity',
                            },
                            {
                                value: 'Rate',
                            },
                            {
                                value: 'Amount',
                            },
                        ],
                        parts,
                        total: [
                            { label: 'Subtotal', value: payload.totalWithoutTax, price: true },
                            {
                                label: `Tax (${payload.tax}%)`,
                                value: payload.total - payload.totalWithoutTax,
                                price: true,
                            },
                            { label: 'Total', value: payload.total, price: true },
                            { label: 'Amount Paid', value: paidAmount, price: true },
                            { label: 'Balance Due', value: payload.balanceDue, price: true },
                        ],
                    },
                    legal,
                },
            },
        });
        invoice.generate();
```

Reply:
Just as a rule of thumb, if you have a function or part of a function that makes an HTTP request, you typically mock that in your tests unless its like an end to end or integration test. 

So if you see a `fetch`, `axios`, or `http` (or an instance of a database client) you usually would mock those functions / files.